docs/developers/improvement-proposals.md

```diff
@@ -0,0 +1,98 @@
+---
+layout: page
+title: Gluten Project Improvement Proposals (GPIP)
+nav_order: 10
+parent: Developer Overview
+---
+
+# Gluten Project Improvement Proposals (GPIP)
+
+The Gluten Project Improvement Proposals doc references [the Spark SPIP documentation](https://spark.apache.org/improvement-proposals.html).
+
+The purpose of a GPIP is to inform and involve the user community in major improvements to the Gluten codebase throughout the development process to increase the likelihood that user needs are met. GPIPs should be used for significant user-facing or cross-cutting changes, not small incremental improvements.
```

Comment:
what does `cross-cutting` mean? Use `cutting-edge` instead?

Reply:
Nit. Please break this one line of statements into multiple lines, which is friendly to reading. Please also check other places.

Reply:
Thanks, addressed, PTAL.